src/main/scala/viper/gobra/frontend/info/implementation/resolution/AmbiguityResolution.scala

```diff
@@ -89,6 +89,7 @@ trait AmbiguityResolution { this: TypeInfoImpl =>
           resolve(e) match {
             case Some(p: ap.FunctionKind) => Some(ap.FunctionCall(p, n.args))
             case Some(p: ap.PredicateKind) => Some(ap.PredicateCall(p, n.args))
+            case _ if exprType(e).isInstanceOf[PredT] => Some(ap.PredExprInstance())
```

Comment:
Here, I would put the expression e into the `ap.PredExprInstance()` (as mentioned in an earlier comment).